applications/models.py

```diff
@@ -143,6 +143,27 @@ def upload_resume(self, resume_file):
         self.resume_upload_date = now_in_utc()
         self.save()
 
+    @transition(
+        field=state,
+        source=AppStates.AWAITING_SUBMISSION_REVIEW.value,
+        target=RETURN_VALUE(AppStates.AWAITING_SUBMISSION_REVIEW.value, AppStates.AWAITING_PAYMENT.value))
+    def approve_submission(self):
+        """Approve application submission"""
+        return (AppStates.AWAITING_SUBMISSION_REVIEW.value
+                if self.has_incomplete_submissions() else AppStates.AWAITING_PAYMENT.value)
+
+    @transition(
+        field=state,
+        source=AppStates.AWAITING_SUBMISSION_REVIEW.value,
+        target=AppStates.REJECTED.value
+    )
+    def reject_submission(self):
+        """Reject application submission"""
+
+    def has_incomplete_submissions(self):
+        """Check if the application is waiting for some submissions to be reviewed"""
+        return self.submissions.filter(review_status__isnull=True).exists()
```

Comment:
This will only work if there is one application step defined for the bootcamp. If there are multiple steps and the user has just submitted something for one of the steps, we would want to set the state to "awaiting submissions" if that first submission is approved. It might be better to flip this function to return True if (a) the user has submitted something for all the steps, and (b) they are all approved.  `applications.api.derive_application_state` does this; not sure if that's something we can abstract into a separate API function.